File: WorldFeed/Server/src/Services/Profile/WorldFeed.Profile.API/Application/DTO/QueryDTO/Cursor/UserCursorQueryResultDTO.cs
namespace WorldFeed.Profile.Application.DTO.QueryDTO.Cursor
{
    using Newtonsoft.Json;
    using WorldFeed.Common.DTO.Cursor;
    using WorldFeed.Common.Public.Models.Interfaces.DTO;
    using WorldFeed.Common.Public.Models.Interfaces.DTO.QueryDTO;
    using WorldFeed.Profile.DTO;

    public class UserCursorQueryResultDTO : BaseCursorQueryDTO<IUserDTO>, IUserCursorQueryResultDTO
    {
        private IUserDTO[] _users;

        [JsonProperty("users")]
        public IUserDTO[] Users
        {
            get => _users ?? new IUserDTO[0];
            set
            {
                _users = value;
                Results = value;
            }
        }

        public override int GetNumberOfObjectRetrieved()
        {
            return Users.Length;
        }
    }
}

File: WorldFeed/Client/src/app/core/Core/DTO/Cursor/UserCursorQueryResultDTO.ts
import {BaseCursorQueryDTO} from "./BaseCursorQueryDTO";
import {IUserDTO} from "../../../Public/Models/Interfaces/DTO/IUserDTO";
import {IUserCursorQueryResultDTO} from "../../../Public/Models/Interfaces/DTO/QueryDTO/IUserCursorQueryResultDTO";
import {UserDTO} from "../UserDTO";

export class UserCursorQueryResultDTO extends BaseCursorQueryDTO<IUserDTO> implements IUserCursorQueryResultDTO {
  private _users: IUserDTO[];

  // [JsonProperty("users")]
  get users(): IUserDTO[] {
    return this._users ?? new UserDTO()[0];   // new IUserDTO[0];
  }

  set users(value: IUserDTO[]) {
    this._users = value;
    super.results = value;
  }

  public getNumberOfObjectRetrieved(): number {
    return this.users.length;
  }
}
